sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobServiceSasSignatureValues.java

```diff
@@ -474,60 +478,111 @@ public BlobServiceSasQueryParameters generateSasQueryParameters(SharedKeyCredent
 
     /**
      * Uses a user delegation key to sign these signature values to produce the proper SAS query parameters.
+     * <p>
+     * If {@link #getVersion()} is not set, then the latest service version is used.
+     * </p>
+     * <p>
+     * The type of SAS query parameters returned depends on the following:
+     * <ol>
+     *     <li>If {@link #getBlobName()} is not set, <b>container SAS</b> query parameters are returned.</li>
+     *     <li>If {@link #getBlobName()} and {@link #getSnapshotId()} are set, <b>blob snapshot</b> SAS query parameters
+     *     are returned.</li>
+     *     <li>If only {@link #getBlobName()} is set, <b>blob SAS</b> query parameters are returned.</li>
+     * </ol>
+     *
+     *  See class level JavaDocs for code snippets.
      *
      * @param delegationKey A {@link UserDelegationKey} object used to sign the SAS values.
      * @return {@link BlobServiceSasQueryParameters}
      * @throws IllegalStateException If the HMAC-SHA256 algorithm isn't supported, if the key isn't a valid Base64
      * encoded string, or the UTF-8 charset isn't supported.
-     * @throws NullPointerException if {@code delegationKey} is null. Or if any of {@code version},
-     * {@code canonicalName}, {@code resource}, {@code expiryTime} or {@code permissions} are null.
+     * @throws IllegalArgumentException if {@link #getPermissions()} contains an invalid character for the SAS resource.
+     * @throws NullPointerException if {@code delegationKey} or {@code account} is null.
+     * @see <a href="https://docs.microsoft.com/rest/api/storageservices/create-user-delegation-sas">Create a user
+     * delegation SAS</a>
      */
-    public BlobServiceSasQueryParameters generateSasQueryParameters(UserDelegationKey delegationKey) {
+    public BlobServiceSasQueryParameters generateSasQueryParameters(UserDelegationKey delegationKey, String accountName) {
```

Comment:
This is a breaking change. Should there be an overload that takes just `delegationKey` too as that was supported earlier?

Reply:
It only worked because the had the `setCanonicalName("")` which allowed users to specify account, container and blob name. This overload is used in all the other languages.

Reply:
@alzimmermsft @rickle-msft thoughts?

Reply:
I'm okay with the change if it means we are going to be consistent with the other languages. SAS token generation is hard and being consistent in API will reduce overhead.